libraries/AP_InertialSensor/AP_InertialSensor_Invensense.h

```diff
@@ -122,6 +122,24 @@ class AP_InertialSensor_Invensense : public AP_InertialSensor_Backend
     // are we doing more than 1kHz sampling?
     bool _fast_sampling;
 
+    // use a sample counter to "timestamp" samples from the sensor
+    // Since the system timebase is not synchronized to the sensor sample clock,
+    // this is the best way to identify missing samples in the log. With this
+    // counter supplied to the notify methods, the SampleUS field becomes an integer
+    // sequence which increments each time a sample is read from the FIFO. As long as
+    // the FIFO is read at a sufficient rate, there should be no dropped samples
+    // between the sensor and the call to notify, with the exception of FIFO corruption:
+    // if corruption is detected, _seqcnt is zeroed to indicate a break in continuity.
+    uint32_t _seqcnt;
+
+#define INVENSENSE_DEBUG 1
```

Comment:
Right, I planned to turn that off or remove the whole thing after development is complete.
Although I'm beginning to think that the sample rate computed here is actually what we might want to add to the IMU log messages, instead of a raw sample count. Since the call to Log_Write_IMU (appears to be) async to the sensor drivers, it would be hard to implement a raw sample counter field for that message.